Add a tutorial some more general context to the README. (#155)
glotzerlab/coxeter: doc/source/quickstart.rst

.. _quickstart:

Quickstart Tutorial
===================

> Once you have :ref:`installed <installing>` **coxeter**, most workflows involve creating an instance of a shape:

Reviewer: ```suggestion
Once you have :ref:`installed <installing>` **coxeter**, most workflows involve creating an instance of a shape, such as a :class:`~.shapes.Polygon`:
```

Author: I slightly modified this, converting `~.shapes.Polygon` to `~coxeter.shapes.Polygon`. Although the former works, I'm not sure how robust/reliable the relative lookup is for a document like this that's not actually within the package.

.. code-block:: python

    >>> import coxeter
    >>> square = coxeter.shapes.Polygon([[0, 0], [1, 0], [1, 1], [0, 1]])

> All shapes may be found in the ``coxeter.shapes`` subpackage and are created in a similar manner.

Reviewer: Check my syntax here -- not sure if this is accurate.
```suggestion
All shapes may be found in the :mod:`coxeter.shapes` subpackage and are created in a similar manner.
```

Author: This is right. I also removed the `:mod:` from your suggestion since coxeter already has a default role of any and will figure this out for us.

For instance, making a sphere requires a radius: ``sphere = coxeter.shapes.Sphere(3)``.
Once you have a shape, you can immediately query it for properties.

.. code-block:: python

    >>> square.vertices
    array([[0., 0., 0.],
           [1., 0., 0.],
           [1., 1., 0.],
           [0., 1., 0.]])
    >>> square.area
    1.0
    >>> square.perimeter
    4.0

> **coxeter** comes with a large range of *shape families*, collections of shapes with standard definitions so that you don't have to parameterize them yourself:

Reviewer: My intention in the second sentence is to provide a link to the API docs for users who want to understand more about this snippet. Check my syntax, I'm trying to show the reference with a full class name with its module path like `coxeter.families.RegularNGonFamily` instead of `RegularNGonFamily`.
```suggestion
The **coxeter** library comes with a range of *shape families*, collections of shapes with standard definitions so that you don't have to parameterize them yourself.
The regular :math:`n`-gon family is such an example, provided as :class:`coxeter.families.RegularNGonFamily`.
```

.. code-block:: python

    >>> hexagon = coxeter.families.RegularNGonFamily.get_shape(6)
    >>> hexagon.vertices.round(2)
    array([[ 0.62,  0.  ,  0.  ],
           [ 0.31,  0.54,  0.  ],
           [-0.31,  0.54,  0.  ],
           [-0.62,  0.  ,  0.  ],
           [-0.31, -0.54,  0.  ],
           [ 0.31, -0.54,  0.  ]])

Part of what makes **coxeter** so powerful is that all shapes are mutable.

> This means that once you have a prototype of a shape it can be modified to fit a specific need.
> For example, the snippet below finds the area of the smallest regular pentagon that contains a unit area equilateral triangle:

Reviewer: ```suggestion
This means that once you have a prototype of a shape, it can be modified to fit a specific need.
For example, the snippet below finds the area of the smallest regular pentagon that contains an equilateral triangle of unit area:
```
